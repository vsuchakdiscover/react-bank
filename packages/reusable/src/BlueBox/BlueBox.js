import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import styles from "./BlueBox.module.scss";

const BlueBox = React.forwardRef((props, ref) => (
  <section
    role="contentinfo"
    aria-label={props["aria-label"] ?? props.header}
    className={cx(styles.root, props.className)}
    id={props.id || undefined}
    data-testid={props.id || undefined}
    ref={ref}
    x-ms-format-detection="none"
  >
    <div className={cx(styles.blueBoxHeader, props.headerClassName)}>
      {props.header}
    </div>
    <div className={cx(styles.blueBoxContent, props.childrenClassName)}>
      {props.children}
    </div>
  </section>
));

BlueBox.propTypes = {
  /** Aria label applied to blue box section tag. Defaults to the Header text if aria-label is not provided. */
  "aria-label": PropTypes.string,

  /** Child markup to display within box */
  children: PropTypes.any.isRequired,

  /** CSS class applied to the content */
  childrenClassName: PropTypes.string,

  /** Class to apply to wrapper div */
  className: PropTypes.string,

  /** Header displayed in the blue bar */
  header: PropTypes.any.isRequired,

  /** CSS class applied to the header */
  headerClassName: PropTypes.string,

  /** HTML ID assigned to the top-level DOM element */
  id: PropTypes.string,
};

BlueBox.displayName = "BlueBox";

export default BlueBox;
