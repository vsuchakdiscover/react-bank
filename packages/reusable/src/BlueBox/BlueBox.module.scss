@import "src/scss/variables";

.blueBoxHeader {
  min-height: 50px;
  padding: 12px 10px 10px;
  background: $accent;
  color: $white;
  font-size: 19px;
  font-family: "Meta Web Bold";
  font-weight: normal;
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: normal;
    margin: 0;
    padding: 0;
    font-size: 19px;
  }
}

.blueBoxContent {
  border: 1px solid $light-gray;
  border-top: none;
  padding: 10px;
  @media #{$tablet} {
    padding: 20px;
  }
}
