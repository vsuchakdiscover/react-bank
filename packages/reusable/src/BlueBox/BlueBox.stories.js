import React from "react";
import BlueBox from "./BlueBox";
import { text } from "@storybook/addon-knobs";

export default {
  title: "Bluebox",
};

export const defaultExample = () => (
  <BlueBox header={text("Header", "Blue Box Header")}>
    So many things can go in here!
  </BlueBox>
);

export const withHeaderTagReset = () => (
  <BlueBox header={<h3>Blue Box Header</h3>}>
    So many things can go in here!
  </BlueBox>
);
