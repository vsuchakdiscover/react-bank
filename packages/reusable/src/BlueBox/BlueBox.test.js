import React from "react";
import BlueBox from "./BlueBox";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BlueBox header="Blue Box Header">So many things can go in here!</BlueBox>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
