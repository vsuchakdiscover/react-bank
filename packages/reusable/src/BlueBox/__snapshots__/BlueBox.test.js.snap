// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  aria-label="Blue Box Header"
  className="root"
  role="contentinfo"
  x-ms-format-detection="none"
>
  <div
    className="blueBoxHeader"
  >
    Blue Box Header
  </div>
  <div
    className="blueBoxContent"
  >
    So many things can go in here!
  </div>
</section>
`;
